### ISSUE TITLE
Segfault when playing a H264 video with V4L2 hardware decoding on Raspberry Pi

### ISSUE THREAD

**Issue by trya** [mpv-player/mpv#10701]
### Important Information

- mpv version 0.34.1 with ffmpeg 5.1.2
- Linux Distribution and Version: custom buildroot system for Raspberry Pi 3 and 4, kernel version is the same as currently on RPI OS (5.15.61, commit 5b775d7293eb75d6dfc9c5ffcb95c5012cd0c3f8)
- Problem has been reproduced with kernel 5.15.32 from RPI OS, and mpv 0.33.1.

### Reproduction steps

Play any H264 video with `--hwdec=v4l2m2m-copy`, whatever the VO (I tried gpu with drm context, X11 with x11-egl, SDL).

### Expected behavior

No crash.

### Actual behavior

A segmentation fault occurs before anything is displayed.

### Log file

- [mpv log](https://github.com/mpv-player/mpv/files/9690512/mpv.log)
- [backtrace of the segfault](https://github.com/mpv-player/mpv/files/9690510/gdb.log)

Looks like the V4L2 context does not or fails to allocate buffers, thus causing the segfault thereafter. I couldn't reproduce this issue neither with current RPI OS (Bullseye) or when converting a H264 video with ffmpeg on buildroot with the V4L2 decoder.

**Comment by sfan5** [comment]
The backtrace points to ffmpeg, is there any reason why you suspect this is a bug in mpv?

**Comment by trya** [comment]
I've had no luck with ffplay, I couldn't get it to work on my buildroot system, so I tried conversion with ffmpeg and V4L2 decoding and it did work. It looked like a playback issue at first, since I only got the segfault with mpv, but now you're pointing that out, it could well be a ffmpeg issue. I will try harder with ffplay and if I can reproduce the crash with it, I'll report the issue to ffmpeg and close this one.

**Comment by trya** [comment]
So, I managed to get ffplay to work, I only had to disable audio playback. And the whole video plays with V4L2 decoding, albeit with no seeking possible. So I got onto the reasons why mpv would crash and not ffplay, and the segfault amounts to whether the first try of packet decoding fails or not, apparently depending on whether the V4L2 decoding device is ready to decode (I'm lost on the details). There is a way to work around that problem, but seeking is still a problem (with both ffplay and mpv).

My bug report for ffmpeg: https://trac.ffmpeg.org/ticket/9957
